lib/msf/base/simple/buffer.rb

```diff
@@ -25,6 +25,8 @@ def self.transform(buf, fmt = "ruby", var_name = 'buf')
       when 'raw'
       when 'num'
         buf = Rex::Text.to_num(buf)
+      when 'hex'
+        buf = Rex::Text.to_hex(buf, '') + "\n"
```

Comment:
jeah i know....if you have other ideas (that doesn't involve blacklisting certain output format for the newline) please let me know :)